qa/tasks/salt_manager.py

```diff
@@ -132,6 +132,16 @@ def all_minions_zypper_status(self):
         self.all_minions_zypper_lu()
         self.all_minions_zypper_ps()
 
+    def all_minions_zypper_up_if_needed(self):
+        """
+        List updates and perform update if needed
+        """
+        zypper_lu = "sudo salt \\* cmd.run \'zypper lu || true\' 2>/dev/null"
```

Comment:
I wonder why salt requires 'sudo' here?

Reply:
I am not entirely sure but iirc it doesn't work without `sudo` and after checking `salt_manager.py` and `deepsea.py` code it kinda proves my point. I ll test it to be 100% sure

Reply:
what exactly have you seen in salt_manager and deepsea?

Reply:
that all `salt` commands use `sudo` for which there should be a valid reason. Anyway I ll test it to be sure and change it if needed.

Reply:
@kshtsk The answer to "why is sudo needed" is: teuthology (afaik always) SSHs to the target node as user `ubuntu`.

Reply:
@smithfarm I know, the question is more like why 'salt' is required sudo? maybe it is some permission issue, I don't see a reason for administrator user to be su - to run provisioning  on remote hosts.

Reply:
@kshtsk You want to discuss here why "salt" commands must be run as root? Isn't that outside the scope of this PR?

Reply:
@smithfarm I don't want to discuss, there is a big difference, between discussion, and question, I don't details of 'ceph-salt' architecture, this is why I'm dropping quick question for those who knows.

Reply:
This line does not invoke "ceph-salt". Salt (SaltStack) is a configuration management system. It expects to be able to make changes anywhere in the systems that it manages, and it needs root privileges for that. I suppose Ansible has the same need and is also run as root in the typical case.

Reply:
as for ansible me always running client side ansible-playback as regular user without sudo, the target machines are using opensuse as client (pretty the same as ubuntu) and it is ansbile responsibility to toggle sudo mode for specific commands, only in that cases when it needed...